src/query/storage/m3/storage.go

```diff
@@ -154,6 +156,34 @@ func (s *m3storage) FetchProm(
 		return storage.PromResult{}, err
 	}
 
+	for _, series := range fetchResult.PromResult.Timeseries {
+		if series == nil {
+			continue
+		}
+
+		rollup := false
+		metricName := ""
+		for _, label := range series.Labels {
+			// Check for both the rollup tag and the metric name label.
+			name := string(label.Name)
+			value := string(label.Value)
+			switch name {
+			case coordmodel.RollupTagName:
```

Comment:
nit: better to make a `[]byte` representation of `coordmodel.RollupTagName` and match the tag name and value against that to avoid the allocs

Reply:
Can do

Reply:
Yeah I wonder if we can also use that for the ByName func arg as well instead of that needing to be a string.